src/mds/Server.cc

```diff
@@ -9787,6 +9793,13 @@ void Server::handle_client_mksnap(MDRequestRef& mdr)
     return;
   }
 
+  // check for number of snapshots;
+  // we don't allow any more if we are already at the limit
+  if (diri->inode.rstat.rsnaps == max_snaps_per_dir) {
```

Comment:
`>=`

Reply:
This also checks recursive snapshots, not the number of snapshots on the given directory.

Reply:
@batrick Please review the updated expression to test for per directory snapshot count.